core/trino-main/src/test/java/io/trino/sql/gen/TestExpressionCompiler.java

```diff
@@ -1993,6 +2021,31 @@ public void run()
         }
     }
 
+    private static class RegularExpressionRunnable
+            implements Runnable
+    {
+        private TrinoException exceptionClause;
+        private final String source;
+        private final String pattern;
+
+        public RegularExpressionRunnable(String source, String pattern)
+        {
+            this.source = source;
+            this.pattern = pattern;
+        }
+
+        @Override
+        public void run()
+        {
+            try {
+                JoniRegexpFunctions.regexpReplace(utf8Slice(source), joniRegexp(utf8Slice(pattern)));
```

Comment:
let's add test for regexplike and like for completeness

Reply:
I added test for regexplike which used the new getMatchingOffset method. JoniRegexpReplaceLambdaFunction used the import getSearchingOffset which had been tested